test/cpp/api/sequential.cpp

```diff
@@ -81,6 +113,32 @@ TEST_F(SequentialTest, PushBackAddsAnElement) {
   ASSERT_EQ(sequential->size(), 2);
   sequential->push_back(M(2));
   ASSERT_EQ(sequential->size(), 3);
+
+  // Test named submodules
+  Sequential sequential_named;
+  ASSERT_EQ(sequential_named->size(), 0);
+  ASSERT_TRUE(sequential_named->is_empty());
+
+  sequential_named->push_back({"linear1", Linear(3, 4)});
+  ASSERT_EQ(sequential_named->size(), 1);
+  ASSERT_EQ(sequential_named->named_children()[0].key(), "linear1");
+  sequential_named->push_back({std::string("linear2"), Linear(3, 4)});
+  ASSERT_EQ(sequential_named->size(), 2);
+  ASSERT_EQ(sequential_named->named_children()[1].key(), "linear2");
+
+  sequential_named->push_back({"shared_m1", std::make_shared<M>(1)});
```

Comment:
are there any tests for mixing named and unnamed push_backs?  What do you expect to happen there?

Reply:
I updated it to test mixing named and unnamed push_backs, it should just work as expected.

Reply:
I don't see where you actually check the names of the unnamed push_backs.

Reply:
Fixed